src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAXIMUM_NUMBER_OF_ELEMENTS = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAXIMUM_NUMBER_OF_ELEMENTS];
+        values = (V[]) new Object[MAXIMUM_NUMBER_OF_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keysAreEquals(key, keys[i])) {
```

Comment:
This can be extracted to a separate method `indexOf(key)` and reused in all places.
```suggestion
        for (int i = 0; i < size; i++) {
            if (keysAreEquals(key, keys[i])) {
                 return i;
             }
```